### ISSUE TITLE
Erreur Google Analytics Production AWS

### ISSUE THREAD

**Issue by zigou13** [jovotech/jovo-framework#894]
## I'm submitting a...

- [x ] Bug report
- [ ] Feature request
- [ ] Documentation issue or request
- [ ] Other... Please describe:

## Expected Behavior
Hello, locally Google Analytics works, but when I put on AWS, I have an error

Here is the code in my config.js

`user: {
    metaData: {
      enabled: true,
      lastUsedAt: true,
      sessionsCount: true,
      createdAt: true,
      requestHistorySize: 4,
      devices: true,
    },
  },
  analytics: {
    // Configuration for generic tracking plugin
    GoogleAnalytics: {
      trackingId: 'UA-18902XXX',
      trackDirectives: true, // Optional
      enableAutomaticEvents: true, // Optional - set to false to disable
      trackEndReasons: false, // Optional - when set to true the custom metrics 1-6 are used to track endReasons
      sessionTimeoutInMinutes: 5, // Optional - default is 5
      skipUnverifiedUser: true, // Optional - default is true
    },
    // Configurations for platform-specific plugins
    GoogleAnalyticsAlexa: {
      trackingId: 'UA-18902XXX'
    },
  },`
### Error log
`{
  "errorType": "TypeError",
  "errorMessage": "Cannot read property 'lastUsedAt' of undefined",
  "trace": [
    "TypeError: Cannot read property 'lastUsedAt' of undefined",
    "    at GoogleAnalytics.saveStartStateAndLastUsed (/var/task/node_modules/jovo-analytics-googleanalytics/dist/src/GoogleAnalytics.js:41:102)",
    "    at Middleware.run (/var/task/node_modules/jovo-core/dist/src/core/Middleware.js:76:30)",
    "    at Middleware.run (/var/task/node_modules/jovo-core/dist/src/core/Middleware.js:54:69)",
    "    at App.handle (/var/task/node_modules/jovo-core/dist/src/core/BaseApp.js:203:46)",
    "    at async App.handle (/var/task/node_modules/jovo-framework/dist/src/App.js:265:9)",
    "    at async Runtime.exports.handler (/var/task/index.js:26:3)"
  ]
}`

## Your Environment

Jovo CLI Version: 3.2.1
* Operating System:
  jovo-analytics-googleanalytics: 3.4.0
  jovo-core: 3.4.0
  jovo-db-filedb: 3.4.0
  jovo-framework: 3.4.0
  jovo-platform-alexa: 3.4.0
  jovo-platform-dialogflow: 3.4.0
  jovo-plugin-debugger: 3.4.0

Thanks for your help

**Comment by NLueg** [comment]
I have the same issue ... It's not possible to use Google-Analytics

**Comment by freiSMS** [comment]
Hi @NLueg and @zigou13 , I am using google analytics for a longer time and never had this issue. I will later update the plugin to prevent an error if lastUsedAt is not set. Nonetheless I would recommend to find the issue here because without lastUsedAt tracking the timeouts for the plugin will not work. For example take the alexa platform: display devices are sending session ended requests a long time after the session ended (sometimes days). This will result in new sessions tracked on google analytics polluting your data.
Could you verify if the metadata lastUsedAt is set for users having this issue? Could this be linked to issues with config stages (do you have an config.staging.js for example)?
Because you are using the standard config for metadata I would try to replace:
```
user: 
    { metaData: 
{ enabled: true, lastUsedAt: true, sessionsCount: true, createdAt: true, requestHistorySize: 4, devices: true, },
 }
```

with 
```
user: 
{ metaData: true }
```

**Comment by NLueg** [comment]
Hello @freiSMS, this doesn't change anything in my case. The problem here is, that `jovo.$user.$metadata` is `undefined`.
I actually don't know why this happens and how I can resolve this.
I had nothing inside my config related to the user. I tried it with this (as mentionend) without any change:

```ts
const config = {
  ...
  analytics: {
    // Configuration for generic tracking plugin
    GoogleAnalytics: {
      trackingId: 'XXXX',
      trackDirectives: true,   // Optional
      enableAutomaticEvents: true, // Optional - set to false to disable
      trackEndReasons: false, // Optional - when set to true the custom metrics 1-6 are used to track endReasons
    },
    GoogleAnalyticsAlexa: {
      trackingId: 'G-JDHG90CJYV',
    },
    GoogleAnalyticsGoogleAssistant: {
      trackingId: 'G-JDHG90CJYV',
    }
  },

  // Just added this for testing purposes. before I don't defined the user here.
  user: {
    metaData: true,
  },
...
};

export = config;
```

**Comment by NLueg** [comment]
Hello @freiSMS, this doesn't change anything in my case. The problem here is, that `jovo.$user.$metadata` is `undefined`.
I actually don't know why this happens and how I can resolve this.
I had nothing inside my config related to the user. I tried it with this (as mentionend) without any change:

```ts
const config = {
  ...
  analytics: {
    // Configuration for generic tracking plugin
    GoogleAnalytics: {
      trackingId: 'XXXX',
      trackDirectives: true,  
      enableAutomaticEvents: true, 
      trackEndReasons: false, 
    },
    GoogleAnalyticsAlexa: {
      trackingId: 'XXX',
    },
    GoogleAnalyticsGoogleAssistant: {
      trackingId: 'XXX',
    }
  },

  // Just added this for testing purposes. before I don't defined the user here.
  user: {
    metaData: true,
  },
...
};

export = config;
```

**Comment by NLueg** [comment]
Hello @freiSMS, thanks for your message. I finally found the issue in my case.
I forgot to add the following to the `config.ts`:

```ts
  user: {
    metaData: {
      enabled: true,
    },
  },
```

I didn't know that this is required by google-analytics. This maybe should be added to the docs?
But after this I still had the issue actually because I removed the database from my config as I thought I wouldn't need it. 
I actually think that this is not the problem the creator of this Issue had, but I just wanted to clarify it.
